### PR TITLE
[core][dashboard] Update log streaming w/o success bit info

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
This PR removes the 1/0 bit info at the beginning of the stream chunk, and decouple file resolution from log streaming. 
With this PR: 
- If log resolution fails (a None filename is resolved): an exception with FileNotFound would be sent to the client. 
- If log resolution succeeds, but streaming fails in the middle: the client should have exceptions thrown automatically as the stream closed. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/experimental/state/api.py [hunk 1]
```diff
@@ -1217,20 +1220,15 @@ def get_log(
         stream=True,
     ) as r:
         if r.status_code != 200:
-            raise RayStateApiException(r.text)
-        for bytes in r.iter_content(chunk_size=None):
-            bytes = bytearray(bytes)
-            # First byte 1 means success.
-            if bytes.startswith(b"1"):
-                bytes.pop(0)
```
Comment (alanwguo) [inline python/ray/experimental/state/api.py:1225]: For removing the first but, can we do it only if a parameter is set to maintain backwards compatibility with old version of ray?

Comment (rickyyx) [reply python/ray/experimental/state/api.py:1225]: Yeah, we could do that.

File: python/ray/experimental/state/api.py [hunk 2]
```diff
@@ -1207,20 +1207,17 @@ def get_log(
...
         suffix=suffix,
         attempt_number=attempt_number,
+        ignore_server_stream_error=None,
     )
-    options_dict = {}
-    for field in fields(options):
-        option_val = getattr(options, field.name)
-        if option_val is not None:
-            options_dict[field.name] = option_val
+    options_dict = options.dict()
```
Comment (rkooo567) [inline python/ray/experimental/state/api.py:1212]:
```suggestion
    options_dict = options.as_dict()
```
nit comment! (same as dataclass I believe)

File: dashboard/modules/state/state_head.py [hunk 3]
```diff
@@ -411,37 +411,61 @@ async def get_logs(self, req: aiohttp.web.Request):
...
+            options.node_id = node_id
+            await response.prepare(req)
+        except Exception as e:
+            response.set_status(500, f"Failed to resolve filename for {options}: {e}")
+            await response.prepare(req)
+            return response
 
         # NOTE: The first byte indicates the success / failure of individual
         # stream. If the first byte is b"1", it means the stream was successful.
         # If it is b"0", it means it is failed.
```
Comment (rkooo567) [inline dashboard/modules/state/state_head.py:443]: This comment needs to be changed?

**Review by rkooo567 — COMMENTED** [review]
Actually why don't we just remove this feature?

**Comment by rickyyx** [comment]
> Actually why don't we just remove this feature?

I think Alan mentioned it would be nice to support back-compatibility (some of dashboard front-end is depending on this bit info I believe) cc @alanwguo

**Comment by alanwguo** [comment]
> > Actually why don't we just remove this feature?
> 
> I think Alan mentioned it would be nice to support back-compatibility (some of dashboard front-end is depending on this bit info I believe) cc @alanwguo

Yes, I believe there is some code on the product that is depending on the current state api logs behavior. @galenhwang , please confirm.

**Comment by galenhwang** [comment]
The dependency can probably safely be removed. Our code could remove the first bit depending on >Ray 2.4 or not if needed (right now it's doing `log_string = log_string[1:]`.

**Comment by rickyyx** [comment]
> The dependency can probably safely be removed. Our code could remove the first bit depending on >Ray 2.4 or not if needed (right now it's doing something like `log_string = log_string[1:]`).

So it's safe for me to simply remove this bit from the OSS? 
I don't think we need to worry too much about other compatibility issues since this is a private endpoint also.

**Comment by galenhwang** [comment]
Confirming product doesn't have a dependency. I'm indifferent to keeping it (some user could be parsing `int(log_string[0])` but I think it's more likely they just do substring so removal is lower risk)

**Comment by rickyyx** [comment]
@alanwguo - Is there any existing dashbaord client that depend on the 1/0 bit? If so - I could change that as well.

**Comment by rkooo567** [comment]
Lmk the conclusion!

**Comment by alanwguo** [comment]
> Confirming product doesn't have a dependency. I'm indifferent to keeping it (some user could be parsing `int(log_string[0])` but I think it's more likely they just do substring so removal is lower risk)

Which is it? There's no dependency, or we need to remove the dependency? I don't really like the idea of doing a ray version check in the code but we could if necessary

**Comment by rickyyx** [comment]
> > Confirming product doesn't have a dependency. I'm indifferent to keeping it (some user could be parsing `int(log_string[0])` but I think it's more likely they just do substring so removal is lower risk)
> 
> Which is it? There's no dependency, or we need to remove the dependency? I don't really like the idea of doing a ray version check in the code but we could if necessary

Synced with @galenhwang offlline - there's no existing such usage within product.

**Comment by rickyyx** [comment]
keep

**Pull request closed without merging**